scripts/nlp_language_modeling/convert_mpt_7b_hf_to_nemo.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+A script to convert the Mosaic MPT-7B checkpoint on HuggingFace to Megatron GPTModel
+This script is hardcoded specifically for the MPT-7B pretrained model only, and is not
+generalisable to any other models.
+
+This script will load and convert the model entirely on CPU for OOM safety, but there
+is an option to put the model onto GPU before the save down, which sets the map_location
+to cuda for the restore_from call. You can do this by adding --cuda to this script call.
+
+This script requires that you have downloaded the 2 .bin weight files for MPT-7B from
+HuggingFace located here: https://huggingface.co/mosaicml/mpt-7b/tree/main
+These files MUST have the following file names and be saved somewhere where this script
+can read them:
+    pytorch_model-00001-of-00002.bin
+    pytorch_model-00002-of-00002.bin
+
+This script will generate a Megatron model with TP=1 and PP=1. If you need different TP/PP
+values, then after running this script, please use the script located below to set whatever
+TP/PP values you want:
+    NeMo/examples/nlp/language_modeling/megatron_change_num_partitions.py
+
+
+Here is an example usage command:
+
+```python
+python scripts/nlp_language_modeling/convert_mpt_7b_hf_to_nemo.py -i /path/to/mpt_7b -o /path/to/save
+```
+
+"""
+
+
+import argparse
+import os
+
+import pytorch_lightning as pl
+import torch
+from omegaconf import OmegaConf
+
+from nemo.collections.nlp.models.language_modeling.megatron import GPTModel
+from nemo.collections.nlp.models.language_modeling.megatron_gpt_model import MegatronGPTModel
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-i', '--input', required=True, type=str, help='path to the two MPT-7B .bin weight files from HuggingFace'
+    )
+    parser.add_argument(
+        '-o', '--output', required=False, default=None, type=str, help='path to dir where to store output .nemo file'
+    )
+    parser.add_argument('--cuda', action='store_true', help='put Nemo model onto GPU prior to savedown')
+
+    args = parser.parse_args()
+
+    if not os.path.exists(args.input):
+        print('ERROR: input directory does not exist or cannot be found. Aborting.', flush=True)
```

Comment:
better to `raise ValueError` here instead of print. We don't use print anywhere in the code base. Also as an FYI, if you need to log anything use logging from nemo.utils

Reply:
I have replaced everything with Nemo logging